marlowe-playground-client/src/Demos/Types.purs

```diff
@@ -0,0 +1,25 @@
+module Demos.Types where
+
+import Prelude
+import Analytics (class IsEvent, Event)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Projects.Types (Lang)
+
+newtype Demo
+  = Demo String
+
+derive instance newtypeDemo :: Newtype Demo _
+
+data Action
+  = LoadDemo Lang Demo
+  | LoadProject
+  | NewProject
+
+defaultEvent :: String -> Event
+defaultEvent action = { category: Just "Demos", action, label: Nothing, value: Nothing }
+
+instance isEventAction :: IsEvent Action where
+  toEvent (LoadDemo lang _) = Just { category: Just "Demos", action: "LoadDemo", label: Just (show lang), value: Nothing }
+  toEvent LoadProject = Just { category: Just "Demos", action: "LoadProject", label: Nothing, value: Nothing }
+  toEvent NewProject = Just { category: Just "Demos", action: "NewProject", label: Nothing, value: Nothing }
```

Comment:
It seems the `defaultEvent` you've defined above would be ideal here.

Reply:
ended up removing those other type constructors anyway